tests/unit/ProductFilter/Options/AttributeOptionsProviderTest.php

```diff
@@ -0,0 +1,133 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License version 3.0
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License version 3.0
+ */
+
+namespace ProductFilter\Options;
+
+use PHPUnit\Framework\TestCase;
+use PrestaShop\Module\PsxMarketingWithGoogle\ProductFilter\Options\AttributeOptionsProvider;
+use PrestaShop\Module\PsxMarketingWithGoogle\Repository\AttributesRepository;
+use PrestaShop\Module\PsxMarketingWithGoogle\Repository\LanguageRepository;
+
+class AttributeOptionsProviderTest extends TestCase
+{
+    public function testTransformRawDataToValidOptions()
+    {
+        $stubAttributeRepository = $this->createStub(AttributesRepository::class);
+        $stubLanguageRepository = $this->createStub(LanguageRepository::class);
+
+        $stubAttributeRepository->method('getCustomAttributesWithLocalizedValues')
+            ->willReturn($this->getInputData());
+        $stubLanguageRepository->method('getIsoById')
+            ->will($this->onConsecutiveCalls('fr', 'en_gb', 'fr', 'en_gb', 'fr', 'en_gb', 'fr', 'en_gb'));
+
+        $provider = new AttributeOptionsProvider(
+            $stubAttributeRepository,
+            $stubLanguageRepository
+        );
+        $result = $provider->getOptions();
+        $this->assertSame(
+            [
+                [
+                    'id' => '2',
+                    'key' => 'Form',
+                    'values' => [
+                        ['id' => '4', 'key' => 'Form', 'value' => 'Classique', 'language' => 'fr'],
+                        ['id' => '4', 'key' => 'Form', 'value' => 'Classic', 'language' => 'en_gb'],
+                        ['id' => '5', 'key' => 'Form', 'value' => 'Chromatique', 'language' => 'fr'],
+                        ['id' => '5', 'key' => 'Form', 'value' => 'Shiny', 'language' => 'en_gb'],
```

Comment:
why is the "key" is part of the values as it seems to be part of the object ?
it seems redundant.